plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -351,9 +351,8 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
       void on_incoming_transaction_async(const transaction_metadata_ptr& trx, bool persist_until_expired, next_function<transaction_trace_ptr> next) {
          chain::controller& chain = chain_plug->chain();
          const auto& cfg = chain.get_global_properties().configuration;
-         signing_keys_future_type future =
-               transaction_metadata::create_signing_keys_future( trx, *_thread_pool, chain.get_chain_id(),
-                     fc::microseconds( cfg.max_transaction_cpu_usage ) );
+         signing_keys_future_type future = transaction_metadata::start_recover_keys( trx, *_thread_pool,
+               chain.get_chain_id(), fc::microseconds( cfg.max_transaction_cpu_usage ) );
          boost::asio::post( *_thread_pool, [self = this, future, trx, persist_until_expired, next]() {
             if( future.valid() )
```

Comment:
Instead of waiting on the future. You could simply call `recover_keys` to ensure it was computed and cached.

Reply:
Would require a mutex.

Reply:
We can wait for the minor release for the larger changes needed to actually make `transaction_metadata` thread-safe.